pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -143,33 +142,37 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 		}
 		// the status map doesn't exist yet, watching below is still an OK thing to do
 	}
+	originalNodes := 0
+	remainingNodes := util.NewSet()
+	checkStatus := func() {
```

Comment:
Could this function take parameters and return values instead of setting external variables? At least in my experience, it makes the code harder to read than if the function has a set of parameters and return variables.